mqtt/mqtt-broker/src/configuration.rs

```diff
@@ -0,0 +1,304 @@
+use config::{Config, ConfigError, File, FileFormat};
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Deserializer};
+use std::path::Path;
+use std::time::Duration;
+
+pub const DEFAULTS: &str = include_str!("../config/default.json");
+
+#[derive(Debug, Deserialize)]
+pub enum QueueFullAction {
```

Comment:
Should we use `#[serde(rename_all = "snake_case")] ` on top of enum instead of specifying each value separately? https://serde.rs/container-attrs.html